autogpt_platform/backend/backend/blocks/hubspot/engagement.py

```diff
@@ -0,0 +1,121 @@
+from datetime import datetime, timedelta
+
+from backend.blocks.hubspot._auth import (
+    HubSpotCredentials,
+    HubSpotCredentialsField,
+    HubSpotCredentialsInput,
+)
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import SchemaField
+from backend.util.request import requests
+
+
+class HubSpotEngagementBlock(Block):
+    class Input(BlockSchema):
+        credentials: HubSpotCredentialsInput = HubSpotCredentialsField()
+        operation: str = SchemaField(
+            description="Operation to perform (send_email, track_engagement)",
+            default="send_email",
+        )
```

Comment:
Why aren't these separate blocks?

Reply:
They don't need to be seperate blocks. I plan to update the user experience though later to show and hide inputs based on what event is being done.

Reply:
I'm not familiar with Hubspot, but they seem two non-overlapping operations to me. So why are they in one block? The same goes for the `HubspotCompanyBlock` and `HubspotContactBlock`, which both support create/update/get operations with non-overlapping implementations.

Merging these in one block is a break of pattern regarding how we implement blocks.